v0/destinations/fb_custom_audience/transform.js

```diff
@@ -187,114 +186,149 @@ function prepareDataField(userSchema, userUpdateList) {
   return data;
 }
 
-function preparePayload(userUpdateList, userSchema, bufferPayload) {
-  const payload = {};
-  Array.isArray(userSchema)
-    ? (payload.schema = userSchema)
-    : (payload.schema = [userSchema]);
-  payload.data = prepareDataField(userSchema, userUpdateList);
-  if (isDefinedAndNotNull(bufferPayload.is_raw)) {
-    if (typeof bufferPayload.is_raw === "boolean") {
-      payload.is_raw = bufferPayload.is_raw;
-    } else {
+function preparePayload(
+  userUpdateList,
+  userSchema,
+  paramsPayload,
+  isHashRequired
+) {
+  const prepareFinalPayload = paramsPayload;
+  const { data_source } = paramsPayload;
+  if (Array.isArray(userSchema)) {
+    prepareFinalPayload.schema = userSchema;
+  } else {
+    prepareFinalPayload.schema = [userSchema];
+  }
+
+  prepareFinalPayload.data = prepareDataField(
+    userSchema,
+    userUpdateList,
+    isHashRequired
+  );
+  if (isDefinedAndNotNull(prepareFinalPayload.is_raw)) {
+    if (typeof prepareFinalPayload.is_raw !== "boolean") {
       logger.debug("is_raw field is not boolean, so dropping it");
+      delete prepareFinalPayload.is_raw;
     }
   }
   if (
-    isDefinedAndNotNull(bufferPayload.type) &&
-    isDefinedAndNotNull(bufferPayload.sub_type) &&
-    typeFields.includes(bufferPayload.type) &&
-    subTypeFields.includes(bufferPayload.sub_type)
+    !(
+      isDefinedAndNotNull(data_source.type) &&
+      isDefinedAndNotNull(data_source.sub_type) &&
+      typeFields.includes(data_source.type) &&
+      subTypeFields.includes(data_source.sub_type)
+    )
   ) {
-    payload.data_source = {
-      type: bufferPayload.type,
-      sub_type: bufferPayload.sub_type
-    };
+    delete prepareFinalPayload.data_source;
   }
-  return payload;
+  return prepareFinalPayload;
 }
 
-function prepareResponse(message, destination, category, userUpdateCategory) {
+function prepareResponse(
+  message,
+  destination,
+  category,
+  userUpdateCategory,
+  isHashRequired
+) {
   const { accessToken, userSchema } = destination.Config;
   const prepareParams = {};
-  let bufferPayload = {};
-  bufferPayload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  let sessionPayload = {};
+  let paramsPayload = {};
+  let dataSsource = {};
+  sessionPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION.name]
+  );
```

Comment:
If this `if condition` is not satisfied we are not adding the session details? So should we generate or something? Or log or something

Reply:
added a log